github/messages.go

```diff
@@ -307,3 +315,144 @@ func ParseWebHook(messageType string, payload []byte) (interface{}, error) {
 	}
 	return event.ParsePayload()
 }
+
+// WebhookTypes returns a list of all the known GitHub event type strings
+// supported by go-github.
+func MessageTypes() []string {
+	types := make([]string, 0, len(eventTypeMapping))
+	for t := range eventTypeMapping {
+		types = append(types, t)
+	}
+	return types
+}
+
+// EventForType returns an empty struct matching the specified GitHub event type.
+// If messageType does not match any known event types, it returns nil.
+func EventForType(messageType string) interface{} {
+	switch messageType {
+	case "branch_protection_rule":
```

Comment:
This thing is such a pain to remember for new PR contributors.

What if we were to update `eventTypeMapping` to include both fields we want to look up?

For example:

```go
type eventTypeTuple struct {
  eventName string
  emptyStruct interface{}
}

var (
  eventTypeMapping = map[string]eventTypeTuple{
    "branch_protection_rule": {"BranchProtectionRuleEvent", &BranchProtectionRuleEvent{}},
    "check_run":              {"CheckRunEvent", &CheckRunEvent{}},
    ...
  }
)
```

Then we could change this `EventForType` to also use the same map.
Finally, every time a new event is added, only one line needs updating in the `eventTypeMapping`.

Thoughts?

Reply:
I thought about this -- we need to make a _copy_ of the object, which means using reflection given that the declared type is `interface{}`.  If the performance cost seems acceptable, that sounds great to me.